lwt-unix/conduit_lwt_unix.ml

```diff
@@ -160,6 +160,13 @@ let init ?src ?(tls_server_key=`None) () =
     | {ai_addr;_}::_ -> Lwt.return { src=Some ai_addr; tls_server_key }
     | [] -> Lwt.fail_with "Invalid conduit source address specified"
 
+let safe_close fd =
+  Lwt.catch
+    (fun () -> Lwt_unix.close fd)
```

Comment:
Can we check with [Lwt_unix.state](http://ocsigen.org/lwt/4.1.0/api/Lwt_unix#VALstate) here before the call the `Lwt_unix.close`?  Then we can avoid the double-close without having to ignore the `EBADF`.

Something like (typed directly into github without testing):
```ocaml
let close_if_open fd =
  match Lwt_unix.state with
  | Lwt_unix.Aborted _
  | Lwt_unix.Opened -> Lwt_unix.close fd
  | Lwt_unix.Closed -> Lwt.return_unit
```